autorest/m4reformatter/__init__.py

```diff
@@ -444,7 +444,13 @@ def update_content_type_parameter(
         return yaml_data
     param = copy.deepcopy(yaml_data)
     param["schema"] = KNOWN_TYPES["string"]  # override to string type
-    param["required"] = False
+    if (
```

Comment:
there should also be a check here where we only do this for version tolerant or low level client